src/utils/inject-stylesheet.js

```diff
@@ -7,13 +11,62 @@
  *
  * @returns {void}
  */
-export function injectStylesheet(endpoint, env, version, uid, cid) {
-	const queryParams = (cid) ? '?previewcid=' + cid : '';
+export function injectStylesheetWithAutoReplace(endpoint, env, version, uid, cid, attempt = 0) {
+	injectStylesheet(endpoint, env, version, uid, cid)
+	const url = getStyleSheetUrl(endpoint, env, version, uid, cid);
+	const cssFile = getCSSFile(url);
 
+	if (cssFile && window.MutationObserver) {
+		const observer = new window.MutationObserver(function (mutationList) {
+			let replaced = false;
+			mutationList.forEach(function (mutation) {
+				mutation.removedNodes.forEach(function (node) {
+					if (node.nodeName === 'LINK' && node.href === url && attempt < MAX_RETRIES && !replaced) {
+						console.warn('Evolv AI: CSS removed, re-adding');
+						observer.disconnect();
+						replaced = true;
+						injectStylesheetWithAutoReplace(endpoint, env, version, uid, cid, attempt + 1);
+					}
+
+					if (attempt === MAX_RETRIES) {
+						console.warn('Evolv AI: re-adding CSS the maximum number of times');
```

Comment:
Technically this block indicates that the max attempts were hit and the css is _not_ being re-added

Reply:
updated language